yt/utilities/lib/misc_utilities.pyx

```diff
@@ -628,11 +628,14 @@ def obtain_relative_velocity_vector(
     cdef np.ndarray[np.float64_t, ndim=3] vzg
     cdef np.ndarray[np.float64_t, ndim=4] rvg
     cdef np.float64_t bv[3]
-    cdef int i, j, k
+    cdef int i, j, k, dim
 
     units = data[field_names[0]].units
     bulk_vector = data.get_field_parameter(bulk_vector).to(units)
-    if len(data[field_names[0]].shape) == 1:
+    dim = len(data[field_names[0]].shape) == 1
```

Comment:
This should be 
```suggestion
    dim = len(data[field_names[0]].shape)
```
also why not using
```suggestion
    dim = data[field_names[0]].ndim
```

Reply:
oh right, absolutely.
I'll try your second suggestion and see if anything goes wrong